subsys/net/lib/app/init.c

```diff
@@ -238,6 +238,11 @@ int net_app_init(const char *app_info, u32_t flags, s32_t timeout)
 		NET_INFO("%s", app_info);
 	}
 
+	if (!iface) {
```

Comment:
This fix is ok, but I wonder if we could compile error if there is no network interface during compile time. I mean it probably does not make sense to continue with networking if there is no network interface defined anywhere.